### DISCUSSION TITLE
How can I extend existing TextMate grammar rules?

### DISCUSSION THREAD

**Discussion by hembhagat99** [microsoft/vscode-discussions#2464] [Extension Development QnA]
Is there a way to extend existing TextMate grammar rules for a language? Example: I want to identify javascript constants with the name in ALL_CAPS. I want to extend the `variable.other.constant.js` scope with the Regex pattern to match all capital letters. I want to call it `variable.other.constant.uppercase.js`.

I tried using `injectTo` as mentioned in this https://github.com/Microsoft/vscode/issues/2793#issuecomment-221093247. But can't make it work.

Below are my `dummy.tmLanguage.json`  and `package.json` files. Please let me know if I am doing something wrong.

**dummy.tmlanguage.json file**
```json
{
    "$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
    "scopeName": "source.js.dummy",
    "name": "Dummy",
    "fileTypes": ["js"],
    "injectionSelector": "variable.other.constant.js",
    "patterns": [
        {
            "name": "variable.other.constant.uppercase.js",
            "match": "\\b[A-Z_][A-Z0-9_]*\\b"
        }
    ]
}
```

**package.json file**
```json
"grammars": [
  {
    "scopeName": "source.js.dummy",
    "path": "./syntaxes/dummy.tmLanguage.json",
    "injectTo": ["source.js"]
  }
]
```

**Comment by RedCMD** [comment]
you can only inject into a `patterns` rule
(`begin`/`end` have an implied `patterns`)

I would assume JavaScripts variable is matched via `match`
so you will need to find a higher scope to inject to
`L:` allows your rule to take priority over JavaScripts
